src/xcode/ENA/ENA/Source/Scenes/QRScanner/QRScannerCoordinator.swift

```diff
@@ -718,6 +686,182 @@ class QRScannerCoordinator {
 		animator.startAnimation()
 	}
 
+	// MARK: Error Alerts
+	
+	private func showTicketValidationErrorAlert(error: TicketValidationError, serviceProvider: String) {
+		let title: String
+		if case .allowListError(.SP_ALLOWLIST_NO_MATCH) = error {
+			title = AppStrings.TicketValidation.Error.serviceProviderErrorNoMatchTitle
+		} else {
+			title = AppStrings.TicketValidation.Error.title
+		}
+		
+		let alert = UIAlertController(
+			title: title,
+			message: error.errorDescription(serviceProvider: serviceProvider),
+			preferredStyle: .alert
+		)
+
+		alert.addAction(
+			UIAlertAction(
+				title: AppStrings.Common.alertActionOk,
+				style: .default,
+				handler: { [weak self] _ in
+					self?.qrScannerViewController?.activateScanning()
+				}
+			)
+		)
+		
+		if case .versionError = error {
+			alert.addAction(
+				UIAlertAction(
+					title: AppStrings.TicketValidation.Error.updateApp,
+					style: .default,
+					handler: { _ in
+						LinkHelper.open(urlString: "https://apps.apple.com/de/app/corona-warn-app/id1512595757?mt=8")
+					}
+				)
+			)
+		}
+
+		DispatchQueue.main.async {
+			self.qrScannerViewController?.present(alert, animated: true)
+		}
+	}
+
+	// swiftlint:disable cyclomatic_complexity
```

Comment:
opt out that check isn't a good solution.
Better refactor that func so it isn't needed.